src/helpers.rs

```diff
@@ -827,6 +833,256 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             None => tcx.item_name(def_id),
         }
     }
+
+    // Call specified external C function, with supplied arguments.
+    // Need to convert all the arguments from their hir representations to 
+    // a form compatible with C (through libffi call).
+    // Then, convert return from the C call into a corresponding form that 
+    // can be stored in MIRI internal memory.
+    fn call_and_add_external_c_fct_to_context(&mut self, 
+        external_fct_defn: ExternalCFuncDeclRep<'tcx>, dest: &PlaceTy<'tcx, Tag>, args: &[OpTy<'tcx, Tag>]) -> InterpResult<'tcx, ()> {
+        
+        let this = self.eval_context_mut();
+        let link_name = external_fct_defn.link_name;
+        use std::ops::Deref;
+
+        // get the function arguments, and convert them to libffi compatible form
+        let mut libffi_args = Vec::<(Box<dyn Any>, CArg)>::with_capacity(args.len());
```

Comment:
Why are we using `Any` at all given that `CArg` can represent all the types you put into `Any`?

Reply:
I swear there was a reason at some point -- but you're right, it's not needed :P